### ISSUE TITLE
Add some documentation about covered guidelines

### ISSUE THREAD

**Issue by kaikreuzer** [openhab/static-code-analysis#23]
@svilenvul I am personally not having the overview about which [bullets from our guidelines](https://www.eclipse.org/smarthome/documentation/development/guidelines.html) are already covered by the tool and which aren't yet.

Would you be able to provide a short info about that in the tool's documentation / README?

This would also be very helpful to track, which new checks might still need to be implemented.

**Comment by petarnv** [comment]
@kaikreuzer, I have opened a PR for adding the coverage of the `ESH Coding Guidelines` in the documentation of the tool(https://github.com/openhab/static-code-analysis/pull/24). Maybe it is good to be noticed that there are checks not only from these guidelines. All the included checks at the moment from `checkstyle`, `findbugs `and `pmd `can be seen in the rulesets (https://github.com/openhab/static-code-analysis/tree/master/src/main/resources/rulesets).

**Comment by kaikreuzer** [comment]
Thanks @petarnv!
Would it make sense to add also information on whether it is a "strict" check, i.e. it makes the build fail? This is something that I would expect for most (if not all) of the checks against the guideline.

> Maybe it is good to be noticed that there are checks not only from these guidelines.

Correct, but only a few ones from findbugs and PMD will be considered as errors and break the build, right? Actually, if those are a pre-requisite for having a PR merged, we should add those checks to the guidelines section, wdyt? (Could be a single new entry, referencing the pmd/findbug problem descriptions).

**Comment by petarnv** [comment]
> Would it make sense to add also information on whether it is a "strict" check

Yes, I will update my PR with information about whether a check is "strict".

> only a few ones from findbugs and PMD will be considered as errors and break the build, right? 

I don't think that any check from findbugs and PMD should break the build.

> we should add those checks to the guidelines section, wdyt?

I think that for now the guidelines are fine. If we find another check, that should break the build, maybe then we can discuss changing the guidelines section.

**Comment by kaikreuzer** [comment]
> I don't think that any check from findbugs and PMD should break the build.

Well, this is how it is today, see the "Usage" documentation:

```
for PMD - the build will fail when a rule with Priority "1" is found;
for Checkstyle - a rule with severity="Error";
for Findbugs - any Matcher with Rank between 1 and 4.
```

And there are [PMD rules with priority 1](https://github.com/openhab/static-code-analysis/blob/master/src/main/resources/rulesets/pmd/rules.xml#L79) in our checks.

**Comment by svilenvul** [comment]
> Well, this is how it is today, see the "Usage" documentation

> for PMD - the build will fail when a rule with Priority "1" is found;
for Checkstyle - a rule with severity="Error";

> And there are PMD rules with priority 1 in our checks.
for Findbugs - any Matcher with Rank between 1 and 4.

Correct, @kaikreuzer, these PMD rules will break the build, I will ask @petarnv to include them in #24.

I find an inaccuracy here now:
> for Findbugs - any Matcher with Rank between 1 and 4.

Should be:
> for Findbugs - Bugs with Rank between 1 and 4 and Priority HIGH

We have difficulties now configuring FindBugs as the Rank and the Priority are part of the detectors source code and we couldn't find a way to override them.
>

**Comment by petarnv** [comment]
@kaikreuzer, if you want have a look at the rulesets for checkstyle https://github.com/openhab/static-code-analysis/blob/master/src/main/resources/rulesets/checkstyle/rules.xml) and PMD (https://github.com/openhab/static-code-analysis/blob/master/src/main/resources/rulesets/pmd/rules.xml) and say if that is the actual priority, that the checks should have.

Take for example this check - https://github.com/openhab/static-code-analysis/blob/master/src/main/resources/rulesets/checkstyle/rules.xml#L40. It verifies if there is any internal package exported in the MANIFEST.MF. Right now it's severity is "warning". Should we make it "error"?

After you say that these are the "correct" priorities, I can update the PR with the documentation with the proper priorities for the checks.

**Comment by kaikreuzer** [comment]
> It verifies if there is any internal package exported in the MANIFEST.MF. Right now it's severity is "warning". Should we make it "error"?

Yes, as this seems to exactly match one of our guidelines.

> After you say that these are the "correct" priorities, I can update the PR with the documentation with the proper priorities for the checks.

I don't see anything that looks obviously bad and I assume you used the levels that are defined by PMD & checkstyle (of which I then additionally assume that they were defined with some thought).

@maggu2810, please also have a look at this and check whether this makes sense for ESH code for you as well.

**Comment by svilenvul** [comment]
After #24 is merged, I assume that the issue can be closed

Please feel free to reopen, if needed.
